enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,538 @@
+  - To allow using the same placement group across multiple availability zones
+  - To allow control of machines/workloads across partitions of a Partition placement group
+- Create Placement Groups based on user configuration when they do not exist
+- Remove Placement Groups when they are no longer required
+- Ensure Placement Groups can be removed during the cluster deprovisioning process
+- Allow appropriate usage of Partition Placement Groups across multiple MachineSets.
+
+### Non-Goals
+
+- Allowing this feature to be leveraged during installation (day 0)
+  - Users will be able to modify worker MachineSets after manifest creation if required but this will not be exposed as
+    a part of the installation configuration before manifest generation
+- Support for Placement Groups for Control Plane Machines
+  - Users may replace their control plane at a later date using future automation if they wish to have their control
+    plane within a placement group
+- Support for moving Machines between placement groups
+
+## Proposal
+
+We will introduce new fields within the `Placement` section of the AWS Machine Provider Config which will allow users
+to specify which placement group to create instances within.
+
+If the placement group does not exist, it will be created based upon the group type and, if applicable partition, count provided.
+
+We will also allow end users to specify which partition their instances should be created within when appropriate.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I would like to use a spread placement group to ensure that my critical
+  workloads are unlikely to suffer from hardware failures simultaneously.
+- As an operator of High-Performance Computing workloads on OpenShift, I would like to use a cluster placement group to
+  ensure that there is
+  minimal latency between instances within my cluster, allowing maximal throughput of my workloads.
+- As an operator of a large scale distributed and replicated data store, I would like to use a partition placement
+  group to ensure that different shards of my data are replicated within different fault domains within the same
+  availability zone.
+- As an OpenShift cluster administrator, I would like unneeded placement groups to be cleaned up so that I am not
+  polluting my environment with resources that are no longer in use.
+
+### API Extensions
+
+The new fields for this enhancement will be limited to the `Placement` struct within the `AWSMachineProviderConfig` in
+the Machine v1beta1 API group.
+
+The changes are outlined below:
+
+```golang
+type Placement struct {
+  // Existing Fields: Region, AvailabilityZone and Tenancy
+  ...
+
+  // GroupName specifies the name of an AWS placement group to create the Machine within.
+  // If the group specified does not exist, it will be created based on the GroupType
+  // parameter.
+  // When a Machine is deleted, if this leaves an empty Placement Group, the group is also
```

Comment:
We should call out in this comment that only Placement Groups created by the cluster will be deleted.